tests/test_json.py

```diff
@@ -451,3 +452,44 @@ def test_union_in_map():
 
     new_records = roundtrip(schema, records)
     assert records == new_records
+
+
+def test_with_dependent_schema():
+    """Tests a schema with dependent schema
+    https://github.com/fastavro/fastavro/issues/418"""
+    dependency = {
+        "type": "record",
+        "name": "Dependency",
+        "namespace": "test",
+        "fields": [{
+            "name": "_name",
+            "type": "string"
+        }]
+    }
+
+    schema = {
+        "type": "record",
+        "name": "Test",
+        "namespace": "test",
+        "fields": [{
+            "name": "_name",
+            "type": "string"
+
+        }, {
+            "name": "_dependency",
+            "type": "Dependency"
+        }]
+    }
+
+    records = [{
+        '_name': 'parent',
+        '_dependency': {
+            '_name': 'child'
+        }
+    }]
+
+    parse_schema(dependency, "/tmp/dir")
```

Comment:
I don't think the second argument here is doing what you expect. The second argument should be `True`/`False` and is an option to `expand` the schema. It's only useful when you want to save off the resulting schema. Also, [as per the docs](https://fastavro.readthedocs.io/en/latest/schema.html#fastavro._schema_py.parse_schema), the option when used should be used as a keyword argument because that will be enforced in the future.

I would just change this and the line below to remove the second argument:

```suggestion
    parse_schema(dependency)
```

Reply:
Thanks for the correction, I think it might be a leftover because I started out with `_load_schema` or some other function that I can't recall at the moment.